src/software/amazon/ion/impl/lite/IonIntLite.java

```diff
@@ -152,24 +163,28 @@ public void setValue(Number value)
         checkForLock();
         if (value == null)
         {
+            _actual_type = IntegerType.INT;
             doSetValue(0, true);
         }
         else
         {
             if (value instanceof BigInteger)
             {
+                _actual_type = IntegerType.BIG_INTEGER;
                 BigInteger big = (BigInteger) value;
                 doSetValue(big);
             }
             else if (value instanceof BigDecimal)
             {
+                _actual_type = IntegerType.BIG_INTEGER;
                 BigDecimal bd = (BigDecimal) value;
                 doSetValue(bd.toBigInteger());
             }
             else
             {
                 // XXX this is essentially a narrowing conversion
```

Comment:
Is this comment something that needs attention?  Have we specified the behavior here properly?